tests/fastapi/test_fastapi.py

```diff
@@ -94,8 +95,11 @@ def test_mozlog_without_correlation_id_middleware(client, caplog):
 
 
 def test_request_id_passed_to_all_log_messages(caplog):
-    handler = logging.StreamHandler()
+    buffer = io.StringIO()
+    handler = logging.StreamHandler(stream=buffer)
```

Comment:
Haven't seen this setup for testing log formatting before 🤔. I wonder if we could achieve this with just `caplog`, but if it works...

Reply:
I can investigate if there's something more elegant.. I wanted to test the whole logging stack up to the formatter